lib/ansible/modules/web_infrastructure/jbosscli.py

```diff
@@ -0,0 +1,219 @@
+      - JBoss Cli command
+  src:
+    required: false
+    description:
+      - File with list of commands
+  cli_path:
+    required: false
+    default: /usr/share/wildfly/bin
+    description:
+      - The location in the filesystem where jboss-cli.sh is located
+  user:
+    required: false
+    description:
+      - Jboss management user
+  password:
+    required: false
+    description:
+      - Jboss management user password
+  server:
+    required: false
+    default: localhost:9990
+    description:
+      - JBoss server or domain controller, with management port
+  verbose:
+    required: false
+    default: false
+    description:
+      - Show the JBoss Cli output, commonly in DMR
+notes:
+  - "jboss-cli.sh need to be running on client host, and $JAVA_HOME/bin is needed in Client $PATH"
+  - ""
+"""
+
+EXAMPLES = """
+# chage scan-interval value, on user wildfly installation
+- jboss:
+    command: /subsystem=deployment-scanner/scanner=default:write-attribute(name=scan-interval,value=6000)
+    cli_path: /home/user/wildfly-10.1.0.Final/bin
+
+#  change ExampleDS datasource user-name, on 192.168.20.55:9990 default installation
+- jboss:
+    command: /subsystem=datasources/data-source=ExampleDS:write-attribute(name=user-name,value=other)
+    server: 192.168.20.55:9990
+
+# Undeploy the hello world application on wildfly server
+- jboss:
+    command: undeploy hello.war
+    server: "{{ ansible_hostname}}:9990"
+"""
+
+RETURN = '''
+---
+changed:
+    description: if jboss cli command change something
+    returned: true
+    type: string
+    sample: 'changed: true'
+command:
+    description: Jboss Cli command
+    returned: JBoss Cli command
+    type: string
+    sample: 'command: "/subsystem=deployment-scanner/scanner=default:write-attribute(name=scan-interval,value=1000)"'
+stdout:
+    description: Jboss Cli command output
+    returned: If verbose JBoss Cli command output, else success or failed
+    type: string
+    sample: '{"outcome" => "success"}'
+Error:
+    description: Jboss Cli command error
+    returned: return error if falied
+    type: string
+    sample: '{"outcome" => "success"}'
+
+'''
+
+import os
+import shutil
+import time
+import re
+import grp
+import platform
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            src=dict(),
+            user=dict(),
+            password=dict(),
+            command=dict(),
+            cli_path=dict(default='/usr/share/wildfly/bin'),
+            server=dict(default='localhost:9990'),
+            verbose=dict(default="False"),
+        ),
+    )
+
+    changed = False
+
+    src = module.params['src']
+    user = module.params['user']
+    password = module.params['password']
+    command = module.params['command']
+    cli_path = module.params['cli_path']
+    server = module.params['server']
+    verbose = module.params['verbose']
+    jsout = None
+
+    if command and src:
```

Comment:
You can use `mutually_exclusive` for `command` and `src`. You can check [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L668) and [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/cloudformation.py#L410)